### DISCUSSION TITLE
spacy.lang.en.tag_map.py not found ?

### DISCUSSION THREAD

**Discussion by sabharish-21** [explosion/spaCy#12517] [Help: Installation] [answered]
When i tried to install spacy and import English languages tag_map, I got the following error,

` from spacy.lang.en.tag_map import TAG_MAP `
` ModuleNotFoundError: No module named 'spacy.lang.en.tag_map' `

tag_map.py is available for some other languages like Korean (`ko`),  Japanese (`ja`)

Do you know where I can find `tag_map.py` for English?

**Comment by adrianeboyd** [comment, marked as answer]
Most of the tag maps under `spacy.lang` were removed for v3.0, see: https://spacy.io/usage/v3#migrating-training-mappings-exceptions

In v3.x, the mapping rules are included in the patterns in the `attribute_ruler` component in the `en_core_web_*` pipelines.

Alternatively, the older `spacy.lang.en.tag_map` mappings are still available in spacy v2.x releases.
